Library/Homebrew/utils/curl.rb

```diff
@@ -255,7 +265,7 @@ def curl_check_http_content(url, url_type, specs: {}, user_agents: [:default],
       end
 
       if url.start_with?("https://") && Homebrew::EnvConfig.no_insecure_redirect? &&
-         !details[:final_url].start_with?("https://")
+         !details[:final_url]&.start_with?("https://")
```

Comment:
```suggestion
         details[:final_url].present? && details[:final_url].start_with?("https://")
```

Reply:
Good catch. Using the safe navigation operator with `details[:final_url]` would return `nil` when `:final_url` isn't part of `details` and this would cause a problem due to the negation. Since `!nil` is `true`, the `redirects back to HTTPS` message would erroneously display in that context.

The suggestion above omits the pre-existing negation but I've made sure to include it in this change.